blocks/v2-applications-carousel/v2-applications-carousel.js

```diff
@@ -0,0 +1,164 @@
+        width: 1200,
+      },
+      {
+        media: '(min-width: 1440px)',
+        width: 1440,
+      },
+      {
+        media: '(min-width: 1920px)',
+        width: 1920,
+      },
+    ],
+  },
+];
+
+/**
+ * Creates the image carousel container.
+ * @param {Array<Object>} carouselCardsData - The processed card data.
+ * @returns {HTMLElement} - The image carousel container element.
+ */
+const createImageCarousel = (carouselCardsData) => {
+  const imagesContainer = createElement('div', {
+    classes: `${blockName}__images`,
+  });
+  const fragment = document.createDocumentFragment();
+
+  carouselCardsData.forEach(({ mobileImageUrl = '', desktopImageUrl = '', title = '', index, isActive = false }) => {
+    const imageConfig = createResponsiveImageConfig(mobileImageUrl, desktopImageUrl);
+    const responsivePicture = createResponsivePicture(imageConfig, true, title, `${blockName}__image-display`);
+
+    const imageWrapper = createElement('div', {
+      classes: `${blockName}__image`,
+      props: {
+        id: `${blockName}-image-${index}`,
+        [`data-${blockName}-index`]: index,
+        'aria-hidden': isActive ? 'false' : 'true',
+        role: 'tabpanel',
+        tabindex: isActive ? '0' : '-1',
+      },
+    });
+
+    imageWrapper.appendChild(responsivePicture);
+    fragment.appendChild(imageWrapper);
+  });
+
+  imagesContainer.appendChild(fragment);
+  return imagesContainer;
+};
+
+/**
+ * Extracts card data for the carousel.
+ * @param {NodeListOf<HTMLElement> | HTMLElement[]} cardElements - The card elements.
+ * @returns {Array<Object>} - The parsed card data.
+ */
+const extractCarouselCardData = (cardElements) =>
+  cardElements.map((card, index) => {
+    if (!card) {
+      return null;
+    }
+
+    return {
+      index,
+      title: card.querySelector('div > p')?.textContent?.trim() || '',
+      iconMarkup: card.querySelector('.icon')?.outerHTML || '',
+      mobileImageUrl: card.querySelector('div > div > p:first-of-type picture img')?.src?.split('?')[0] || '',
+      desktopImageUrl: card.querySelector('div > div > p:last-of-type picture img')?.src?.split('?')[0] || '',
+      isActive: index === 0,
+    };
+  });
+
+/**
+ * Creates the navigation container for the carousel.
+ * @param {Array<Object>} carouselCardsData - The processed card data.
+ * @returns {HTMLElement} - The navigation container element.
+ */
+const createCarouselCardsContainer = (carouselCardsData) => {
+  const navigation = createElement('nav', {
+    classes: `${blockName}__cards`,
+    props: {
+      role: 'tablist',
+      'aria-label': 'Applications Carousel Navigation',
+    },
+  });
+  const cardList = createElement('ul', {
+    classes: `${blockName}__cards-list`,
+  });
+  const fragment = document.createDocumentFragment();
+
+  carouselCardsData.forEach(({ title, index, isActive, iconMarkup }) => {
+    const cardItem = createCarouselCard(title, index, isActive, iconMarkup);
+    fragment.appendChild(cardItem);
```

Comment:
Instead of using `stringToElement` to parse the text to html, what about instead of appendChild use `insertAdjacentHTML`?
If you had to do any operation on the html inside that would make more sense... but just to append you don't need to preprocess it...

Reply:
I've removed `stringToElement` and use `insertAdjacentHTML` instead.